### ISSUE TITLE
if i have upside down sequence images, will they flip after upload?

### ISSUE THREAD

**Issue by hughdavenport** [mapillary/mapillary_issues#2327]
On android, my sequence for this mornings ride was upside down even though I'm sure the text was the right way up. Will they be converted on upload or should i do it manually?

Side note, maybe another issue/feature req. Would it be possible to force th screen to changr orientation? My bike mount is set to be one way and to get it to rotate i have to pick up bike sideways and put back down for it to understand what "up" is. On android

**Comment by majkaz** [comment]
Ditto... and few other points:
The android app is ignoring the phone screen orientation setting, too. I have tried everything possible, including setting the screen orientation prior to starting the app. The app quite often decides to go with upside-down orientation and due to the tilt detection refuses take photos at all. Today, I couldn't take photos for 5 km and was fiddling with the app on every red light trying to get it working. Both android 3.0 and 3.01 have this bug.

**Comment by jfburke3** [comment]
@hughdavenport They do appear to be oriented correctly after upload.  Most of the time, anyway.

**Comment by hughdavenport** [comment]
Hmm not mine after uploading. https://www.mapillary.com/app/user/hughdavenport?lat=-41.23401719999998&lng=174.90359205000004&z=17&pKey=TwEkKXGkaPrrLwt0d4Ua7w&focus=photo

**Comment by hughdavenport** [comment]
Happened again, after making sure it was the correct way before clicking start... Kinda feels a bit pointless going on an hour bike ride for the purpose of mapping if it doesn't even automatically correct it after upload, or have a way to do it without manually pulling off phone then manually uploading.

I've only had it happen after the upgrade to 3.0, and today it was on 3.01 so that didn't fix it.

**Comment by hughdavenport** [comment]
And now I managed to get sideways images (portrait). Phone definately not portrait as is in a horizontal mount...

Really feels like I'm wasting my time contributing. And it isn't even open source so i can't even help debug

**Comment by hughdavenport** [comment]
So it looks like that it is preferring the android rotation for saving but displaying its own rotation when it is running (the stop/start button text). Prehaps a way to force a certain alignment (or reverse what alignment it uses) to allow for different mounts. I could rebuild my mount to show the other way but it is a bit of a pain

**Comment by hughdavenport** [comment]
So if i have get it in a point where it is upside down compared to the android (android pull down is at bottom, and the start/stop button text is upside down) thrn the images are correct on upload, BUT it resets which way the start/stop show whenever it tilts slightly so wont be suitable for mount

**Comment by hughdavenport** [comment]
Ok so if i start it when its the right way up. Then flip it the stop button text doesn't flip and seems to show the correct orientation when uploading

**Comment by hughdavenport** [comment]
Ok so updated to 4.0. Now doing the above trick shows phone tilted and no photos taken. Even with the pause when tilted option turned off. Anyone from mapillary going to reply and help debug? Or should i just keep uploaded upside down images? Or just stop contributing?

**Comment by hughdavenport** [comment]
New mount. Phone horizontal. Pictures show up portrait

**Comment by hughdavenport** [comment]
3.04 looks like it might have fixed this

**Comment by hughdavenport** [comment]
3.04 looks fairly good. Its only been upside down once which luckily was a shortish sequence but did cover some untouched areas.

Thought for a feature, deleting a sequence on phone before uploading all? Only way i can see that happening is deleting image by image. Or just the folder in the filesystem

**Comment by peterneubauer** [comment]
Three things here:

* We are thinking about an editor before upload that could fix location and possibly rotation in the app (we are testing it on iOS in the next release)

* On the backend, we are right now testing automatic recognition (via deep learning) of upside down uploads and suggestions to the author to rotate them (with reprocessing).

* In the editor, we want to add the possibility to rotate an image/the whole sequences and thus manually trigger the change and reprocessing.

**Comment by peterneubauer** [comment]
The orientation changing is now implemented in the backend, as a first measure we will let a bot suggest orientation changes and make changesets for them, in the next phase you can in the editor rotate images.

**Comment by randoogle** [comment]
Any progress on this? I'm getting pretty discouraged when about _half_ my sequences end up upside down. They're practically useless for mapping. Got a new phone recently, and was hopeful that I might see an improvement, but it's still the same. I have noticed that they are more often upside down when I don't have a chance to move the phone around before putting it in the mount. The capture interface is always upside down, BTW (I have my phone tilted to the right, if I could physically tilt it left, the interface would be right side up, and I suspect more of my sequences would be right-side up as well; however, where I have it mounted doesn't allow for this).

I think one thing that would really help, in the meantime, is an interface on the website where we could flag our sequences as upside down. I would think that automatically rotating them would be trivial on your end.

**Comment by hughdavenport** [comment]
I wouldn't hold your breath with these devs

**Comment by peterneubauer** [comment]
@hughdavenport I rotated https://www.mapillary.com/app/?lat=-41.23420390000001&lng=174.90374405&z=17&pKey=0u2O9jQQEEMpqYYNWcS_EQ&focus=photo, should take some hours to get through.

@randoogle what phone, can you maybe enable settings->Developer->trace, and send the logs after one of these upside down sequences? Are you biking and shaking the phone?

**Comment by hughdavenport** [comment]
Still very much upside down, but did see that a bot did something on some of my feeds. Pity I don't use this project anymore

**Comment by randoogle** [comment]
It's a Moto G5 Plus (old phone was a Moto E 2nd Gen). And, no, I'm not biking or shaking the phone (at least not very often). In fact, it seems less likely to happen if I shake the phone a bit before starting a new sequence. I'll see about sending a stack trace next time it happens.

**Comment by peterneubauer** [comment]
Thanks!

@hughdavenport investigating your image rotation issue.

**Comment by ovivoda** [comment]
_mapillary_issues is no longer an active repository._

_Let’s continue discussing features in our forum! We will import this issue as topic to the <a href="http://forum.mapillary.com ">Mapillary Forum</a> in the coming weeks. We’ll notify here once it’s been done. See you there!_
